numalogic/registry/redis_registry.py

```diff
@@ -0,0 +1,230 @@
+                serialized_artifact,
+                artifact_version,
+                artifact_timestamp,
+                serialized_metadata,
+            ) = self.client.hmget(
+                name=model_key, keys=["artifact", "version", "timestamp", "metadata"]
+            )
+            deserialized_artifact = loads(serialized_artifact)
+            deserialized_metadata = None
+            if serialized_metadata:
+                deserialized_metadata = loads(serialized_metadata)
+        except RedisError as err:
+            raise RedisRegistryError(f"{err.__class__.__name__} raised") from err
+        else:
+            return ArtifactData(
+                artifact=deserialized_artifact,
+                metadata=deserialized_metadata,
+                extras={
+                    "timestamp": artifact_timestamp.decode(),
+                    "version": artifact_version.decode(),
+                },
+            )
+
+    def save(
+        self,
+        skeys: KEYS,
+        dkeys: KEYS,
+        artifact: artifact_t,
+        **metadata: META_T,
+    ) -> Optional[str]:
+        """
+        Saves the artifact into redis registry and updates version.
+        Args:
+            skeys: static key fields as list/tuple of strings
+            dkeys: dynamic key fields as list/tuple of strings
+            artifact: primary artifact to be saved
+            metadata: additional metadata surrounding the artifact that needs to be saved
+
+        Returns:
+            model version
+        """
+        key = self.construct_key(skeys, dkeys)
+        production_key = self.__construct_production_key(key)
+        version = 0
+        try:
+            if self.client.exists(production_key):
+                _LOGGER.debug("Production key exists for the model")
+                version_key = self.client.get(name=production_key)
+                version = int(self.get_version(version_key.decode())) + 1
+            with self.client.pipeline() as pipe:
+                new_version_key = self.__save_artifact(pipe, artifact, metadata, key, str(version))
+                pipe.expire(name=new_version_key, time=self.ttl)
+                _LOGGER.info("Model is successfully with the key = %s", new_version_key)
+                pipe.execute()
+        except RedisError as err:
+            raise RedisRegistryError(f"{err.__class__.__name__} raised") from err
+        else:
+            return str(version)
+
+    def delete(self, skeys: KEYS, dkeys: KEYS, version: str) -> None:
+        """
+        Deletes the model version from registry.
+        Args:
+            skeys: static key fields as list/tuple of strings
+            dkeys: dynamic key fields as list/tuple of strings
+            version: model version to delete
+        """
+        key = self.construct_key(skeys, dkeys)
+        del_key = self.__construct_version_key(key, version)
+        try:
+            if self.client.exists(del_key):
+                self.client.delete(del_key)
```

Comment:
Can't we just try to delete it directly?

Reply:
Delete does not return anything if the key is present or not.
So, checking it before for better user interface.